src/read.rs

```diff
@@ -591,13 +539,18 @@ impl<R> WavReader<R>
         Ok(spec_ex)
     }
 
-    /// Attempts to create a reader that reads the WAVE format.
-    ///
-    /// The header is read immediately. Reading the data will be done on
-    /// demand.
-    pub fn new(mut reader: R) -> Result<WavReader<R>> {
-        try!(read_wave_header(&mut reader));
-        let (spec_ex, data_len) = try!(read_until_data(&mut reader));
+
+    pub fn into_inner(self) -> R {
+        self.reader
+    }
+
+    pub fn into_wav_reader(mut self) -> Result<WavReader<R>> {
```

Comment:
I’m not sure how useful this is, because of the concerns you raised before about accessing chunks after the DATA chunk. What we could do instead, is allowing to construct the samples iterator from a `WavSpecEx` and a `Reader` that is assumed to be at the start of a DATA chunk. The reader could be an `EmbeddedReader` from `Chunk::Data`. Then if you want access to all chunks and still read the audio data in one pass, you can do so, albeit with a bit more manual work. And if you just want the audio `WavReader` still works, or if you just want the raw data of a specific chunk, `ChunkReader` works. What do you think?